pallets/moonbeam-orbiters/src/lib.rs

```diff
@@ -0,0 +1,485 @@
+	pub enum Error<T> {
+		/// The collator is already added in orbiters program.
+		CollatorAlreadyAdded,
+		/// This collator is not in orbiters program.
+		CollatorNotFound,
+		/// There are already too many orbiters associated with this collator.
+		CollatorPoolTooLarge,
+		/// There are more collator pools than the number specified in the parameter.
+		CollatorsPoolCountTooLow,
+		/// The minimum deposit required to register as an orbiter has not yet been included in the
+		/// onchain storage
+		MinOrbiterDepositNotSet,
+		/// This orbiter is already associated with this collator.
+		OrbiterAlreadyInPool,
+		/// Orbiter cant leave this round
+		OrbiterCantLeaveThisRound,
+		/// This orbiter has not made a deposit
+		OrbiterDepositNotFound,
+		/// This orbiter is not found
+		OrbiterNotFound,
+	}
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(crate) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// Paid the orbiter account the balance as liquid rewards.
+		OrbiterRewarded {
+			account: T::AccountId,
+			rewards: BalanceOf<T>,
+		},
+		/*SelectedOrbiters {
+			orbiters:
+		}*/
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Add an orbiter in a collator pool
+		#[pallet::weight(500_000_000)]
+		pub fn collator_add_orbiter(
+			origin: OriginFor<T>,
+			orbiter: <T::Lookup as StaticLookup>::Source,
+		) -> DispatchResult {
+			let collator = ensure_signed(origin)?;
+			let orbiter = T::Lookup::lookup(orbiter)?;
+
+			let mut collator_pool =
+				CollatorsPool::<T>::get(&collator).ok_or(Error::<T>::CollatorNotFound)?;
+			let orbiters = collator_pool.get_orbiters();
+			ensure!(
+				(orbiters.len() as u32) < T::MaxPoolSize::get(),
+				Error::<T>::CollatorPoolTooLarge
+			);
+			if orbiters.iter().any(|orbiter_| orbiter_ == &orbiter) {
+				return Err(Error::<T>::OrbiterAlreadyInPool.into());
+			}
+
+			// Make sure the orbiter has made a deposit. It can be an old orbiter whose deposit
+			// is lower than the current minimum (if the minimum was lower in the past), so we just
+			// have to check that a deposit exists (which means checking that the deposit amount
+			// is not zero).
+			let orbiter_deposit =
+				T::Currency::reserved_balance_named(&T::OrbiterReserveIdentifier::get(), &orbiter);
+			ensure!(
+				orbiter_deposit > BalanceOf::<T>::zero(),
+				Error::<T>::OrbiterDepositNotFound
+			);
+
+			collator_pool.add_orbiter(orbiter);
+			CollatorsPool::<T>::insert(collator, collator_pool);
+
+			Ok(())
+		}
+
+		/// Remove an orbiter from a collator pool
+		#[pallet::weight(500_000_000)]
+		pub fn collator_remove_orbiter(
+			origin: OriginFor<T>,
+			orbiter: <T::Lookup as StaticLookup>::Source,
+		) -> DispatchResult {
+			let collator = ensure_signed(origin)?;
+			let orbiter = T::Lookup::lookup(orbiter)?;
+
+			let mut collator_pool =
+				CollatorsPool::<T>::get(&collator).ok_or(Error::<T>::CollatorNotFound)?;
+
+			if !collator_pool.remove_orbiter(&orbiter) {
+				Err(Error::<T>::OrbiterNotFound.into())
```

Comment:
should make this `ensure` and write `collator_pool` to storage after it changes in the successful case